Fix reconnecting overlay nodes inside overlays

jorenbroekema opened the pull request:
WIP: doing some tests on lion for a bug we encountered where we have a tooltip inside an element that gets moved to global overlays container (because this element is a global overlay). The disconnectedCallback in overlay mixin is pretty destructive in tearing down stuff for the tooltip when it gets moved, and after the teardown and reconnect, the tooltip is very broken.

This "hacky fix" as I have it now fixes it by reassembling the moved overlay's content and calling setupOverlay again.

I have yet to find out why I need the second hack to fix the arrow in tooltip, and still need to find out what a proper fix would look like.

tlouisse commented on packages/overlays/test-suites/OverlayMixin.suite.js:
```diff
@@ -137,4 +140,59 @@ export function runOverlayMixinSuite({ /* tagString, */ tag, suffix = '' }) {
+        expect(contentNode.innerText).to.equal('content of the nested overlay');
+      }
+
+      expect(true).to.be.true;
```
Not so sure about this one... :D

tlouisse replied:
Or is it meant to make the test pass in case it's a local overlay? In that case, we can just drop this expect and it will pass as well

jorenbroekema replied:
Oops :D

jorenbroekema replied:
removed it

tlouisse commented:
Great job fixing this hard bug 👍 

I think we still agree that it feels like tech debt.
Hopefully we can mitigate this tech debt when we implement this: 
https://github.com/ing-bank/lion/pull/582

Could you maybe add a TODO about this, plus an explanation in the code why all of this is needed (or maybe mention a reference to the test at least)
Then I will approve

jorenbroekema commented:
Can you point out where you'd like to see another TODO? Because I added a TODO, and I added comments in the code why we have a promise->resolve thing in the connectedCallback that awaits updateComplete
